packages/components/src/segmented-control/segmented-control-option.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useInstanceId } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	PolymorphicComponentProps,
+} from '../ui/context';
+import SegmentedControlButton from './segmented-control-button';
+import type {
+	SegmentedControlOptionProps,
+	SegmentedControlContextProps,
+} from './types';
+import { useSegmentedControlContext } from './segmented-control-context';
+
+function getShowSeparator(
+	segmentedControlContext: SegmentedControlContextProps,
+	index: number
+) {
+	const { currentId, items } = segmentedControlContext;
+	const isLast = index === items.length - 1;
+	const isActive = items[ index ]?.id === currentId;
+	const isNextActive = items[ index + 1 ]?.id === currentId;
+
+	let showSeparator = true;
+
+	if ( items.length < 3 ) {
+		showSeparator = false;
+	}
+
+	if ( isActive || isNextActive || isLast ) {
+		showSeparator = false;
+	}
+
+	return showSeparator;
+}
+
+function SegmentedControlOption(
+	props: PolymorphicComponentProps< SegmentedControlOptionProps, 'input' >,
+	forwardedRef: import('react').Ref< any >
+) {
+	const id = useInstanceId( SegmentedControlOption );
```

Comment:
It looks like this is generating invalid ids (ids starting with a number aren't valid). I guess we should address this in the `useInstanceId` function itself at some point (like defining a default prefix). But to fix it in this PR, we can either pass a prefix as the second argument or append it to another string.

Aside from this, I don't see other problems in this PR and I think it's good to go. We can iterate and improve it if needed before it's stable.

Reply:
Good catch! Thanks!